### PR TITLE
Documentation for KStream binder

### LINKED ISSUE #160: Add documentation for KStream binder

### PR DESCRIPTION
**Pull request by sobychacko**
Fix #160

This PR requires the merging of https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/pull/159

### CODE REVIEW

File: spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/overview.adoc [hunk 1]
```diff
@@ -429,4 +430,69 @@ On the other hand, if auto topic creation is disabled on the server, then care m
 
 If you want to have full control over how partitions are allocated, then leave the default settings as they are, i.e. do not exclude the kafka broker jar and ensure that `spring.cloud.stream.kafka.binder.autoCreateTopics` is set to `true`, which is the default.
 
+== Kafka Stream Binding Capabilities of Spring Cloud Stream
```
Comment (artembilan) [inline spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/overview.adoc]: I think the  feature is really called `Kafka Streams`.

File: spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/overview.adoc [hunk 2]
```diff
@@ -429,4 +430,69 @@ On the other hand, if auto topic creation is disabled on the server, then care m
...
 
+Spring Cloud Stream Kafka support also includes a binder specifically designed for Kafka Streams binding.
+Using this binder, applications can be written that leverage the Kafka Streams API.
+For more information on Kafka Streams, see https://kafka.apache.org/0110/documentation/streams/developer-guide[Kafka Streams API Developer Manual]
```
Comment (artembilan) [inline spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/overview.adoc]: Looks like we can have versionless link: https://kafka.apache.org/documentation/streams/developer-guide

File: spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/overview.adoc [hunk 3]
```diff
@@ -429,4 +430,69 @@ On the other hand, if auto topic creation is disabled on the server, then care m
...
+
+[source,xml]
+----
+<dependency>
+  <groupId>org.springframework.cloud</groupId>
+  <artifactId>spring-cloud-stream-binder-kstream</artifactId>
+</dependency>
+----
+
+In addition to leveraging the Spring Cloud Stream programming model which is based on Spring Boot, one of the main other benefits that the Kstream binder provides is the fact that it avoids the boilerplate configuration that one needs to write when using the Kafka Streams API directly.
```
Comment (artembilan) [inline spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/overview.adoc]: I think should be really `KStream`

File: spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/overview.adoc [hunk 4]
```diff
@@ -429,4 +430,69 @@ On the other hand, if auto topic creation is disabled on the server, then care m
...
+
+[source]
+----
+java -jar uber.jar  --spring.cloud.stream.bindings.input.destination=words --spring.cloud.stream.bindings.output.destination=counts
+----
+
+This means that the application will listen from the incoming Kafka topic words and write to the output topic counts.
+
+Spring Cloud Stream will ensure that the messages from both the incoming and outgoing topics are bound as KStream objects.
+As one may observe, the developer can exclusively focus on the business aspects of the code, i.e writing the logic required in the processor rather than setting up the streams specific configuration required by the Kafka Streams infrastructure.
```
Comment (artembilan) [inline spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/overview.adoc]: A period after `e`: `i.e.`

File: spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/overview.adoc [hunk 5]
```diff
@@ -429,4 +430,69 @@ On the other hand, if auto topic creation is disabled on the server, then care m
...
+All those boilerplate is handled by Spring Cloud Stream behind the scenes.
+
+=== Kafka Streams properties
+
+The properties must be prefixed with `spring.cloud.stream.kstream.binder.`.
+
+streamConfiguration::
+  Map with a key/value pair containing properties pertaining to Kafka Streams API.
+  For all the properties that may go into streams configuration, see http://www.trieuvan.com/apache/kafka/0.10.2.1/javadoc/org/apache/kafka/streams/StreamsConfig.html[StreamsConfig javadoc].
```
Comment (artembilan) [inline spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/overview.adoc]: Yeah... Unfortunately we can't have this without version. 
How about just say: 

> see `StreamsConfig` JavaDocs for more information

?

File: spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/overview.adoc [hunk 6]
```diff
@@ -429,4 +430,69 @@ On the other hand, if auto topic creation is disabled on the server, then care m
...
+				.toStream()
+				.map((w, c) -> new KeyValue<>(null, "Count for " + w.key() + ": " + c));
+	}
+
+	public static void main(String[] args) {
+		SpringApplication.run(WordCountProcessorApplication.class, args);
+	}
+----
+
+If you build it as spring boot runnable fat jar, you can run the above example in the following way:
```
Comment (artembilan) [inline spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/overview.adoc]: Must be `Spring Boot`

File: spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/overview.adoc [hunk 7]
```diff
@@ -430,11 +430,13 @@ On the other hand, if auto topic creation is disabled on the server, then care m
...
 
-== Kafka Stream Binding Capabilities of Spring Cloud Stream
+== Kafka Streams Binding Capabilities of Spring Cloud Stream
 
 Spring Cloud Stream Kafka support also includes a binder specifically designed for Kafka Streams binding.
 Using this binder, applications can be written that leverage the Kafka Streams API.
-For more information on Kafka Streams, see https://kafka.apache.org/0110/documentation/streams/developer-guide[Kafka Streams API Developer Manual]
+For more information on Kafka Streams, see https://kafka.apache.org/documentation/streams/developer-guide[Kafka Streams API Developer Manual]
+
+Kafka Streams support in Spring Cloud Stream is based on the foundations provided by the Spring Kafka project. For details on that support, see http://docs.spring.io/spring-kafka/docs/1.1.6.RELEASE/reference/html/_reference.html#kafka-streams[Kafaka Streams Support in Spring Kafka].
```
Comment (artembilan) [inline spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/overview.adoc]: The rule about links: avoid versions as much as possible.
There is http://docs.spring.io/spring-kafka/reference/html/_reference.html#kafka-streams alias pointing to the latest version of the project. This way, when you upgrade and go ahead with your new version, there won't be need to take a look into the docs to fix old versions mentionings

File: spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/overview.adoc [hunk 8]
```diff
@@ -429,4 +430,76 @@ On the other hand, if auto topic creation is disabled on the server, then care m
...
+=== Support for interactive queries
+
+If access to the `KafkaStreams` is needed for interactive queries, the internal `KafkaStreams` instance can be accessed via `KStreamBuilderFactoryBean.getKafkaStreams()`.
+You can autowire the `KStreamBuilderFactoryBean` instance provided by the KStream binder. Then you can get `KafkaStreams` instance from it and retrieve the underlying store, execute queries on it, etc.
+
+=== Kafka Streams properties
+
+The properties must be prefixed with `spring.cloud.stream.kstream.binder.`.
+
+streamConfiguration::
```
Comment (mbogoevici) [inline spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/overview.adoc]: This need to change to `configuration`.

**Pull request merged**